lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -1534,8 +1534,10 @@ bool checkTypeForSPIRVExtendedInstLowering(IntrinsicInst *II, SPIRVModule *BM) {
   case Intrinsic::maxnum:
   case Intrinsic::nearbyint: {
     Type *Ty = II->getType();
-    if (II->getArgOperand(0)->getType() != Ty)
-      return false;
+    for (int Index = 0; Index < II->getNumArgOperands(); ++Index) {
+      if (II->getArgOperand(Index)->getType() != Ty)
+        return false;
+    }
```

Comment:
Probably this check is redundant, I'm not sure it is possible to create these intrinsics with inconsistent argument types.